event-producer/main.py

```
@@ -0,0 +1,206 @@
+    
+    def stop(self):
+        """Signal to stop polling"""
+        self.running = False
+        
+    def __enter__(self):
+        return self
+    
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+def signal_handler(signum, frame):
```

Ruff (0.14.10)

Unused function argument: `frame`

(ARG001)